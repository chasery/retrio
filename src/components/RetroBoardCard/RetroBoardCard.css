.RetroBoardCard {
  transform: scale(1);
  align-self: start;
  list-style: none;
  box-shadow: 0 0.0625rem 0.0625rem #0000001c;
  background-color: #fff;
  transition: all 0.15s cubic-bezier(0.22, 1, 0.36, 1);
}
.RetroBoardCard:hover,
.RetroBoardCard:focus {
  position: relative;
  z-index: 9;
  transform: scale(1.05);
  box-shadow: 0 0.25rem 0.25rem #0000001c;
  transition: all 0.3s cubic-bezier(0.22, 1, 0.36, 1);
}
.RetroBoardCard + .RetroBoardCard {
  margin-top: 1rem;
}

.RetroBoardCard__headline {
  display: flex;
  padding: 0;
  border-bottom: 0.0625rem solid #ebebeb;
}
.RetroBoardCard__headline > * {
  margin: 0;
  padding: 1rem;
}
.RetroBoardCard__headline > h4 {
  flex-grow: 1;
}

.RetroBoardCard__control {
  display: flex;
  position: relative;
  justify-content: flex-end;
  flex-basis: 1rem;
  font-size: 1.5rem;
  font-weight: bold;
  line-height: 1rem;
}
.RetroBoardCard__control > button {
  all: unset;
}

.RetroBoardCard__text.--no-headline {
  display: flex;
  flex-wrap: nowrap;
  position: relative;
}
.RetroBoardCard__text.--no-headline .RetroBoardCard__control {
  position: absolute;
  right: 0;
  padding: 1rem;
}
.RetroBoardCard__text pre {
  padding: 0 1rem;
  font-family: 'Roboto', sans-serif;
  font-weight: 400;
  line-height: 1.5rem;
  word-break: break-word;
  white-space: pre-wrap;
}

.RetroBoardCard__creator {
  margin: 0;
  padding: 0.75rem 1rem;
  color: #8b8b8b;
  font-size: 0.75rem;
}
