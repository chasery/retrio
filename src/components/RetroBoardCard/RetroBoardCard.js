import React from 'react';
import './RetroBoardCard.css';

function RetroBoardCard(props) {
  const { headline, text, submitted_by } = props;

  return (
    <li className='RetroBoardCard'>
      <h4 class='RetroBoardCard__headline'>{headline}</h4>
      <div class='RetroBoardCard__text'>
        <p>{text}</p>
      </div>
      <div class='RetroBoardCard__creator'>
        Submitted by <strong>{submitted_by}</strong>
      </div>
    </li>
  );
}

export default RetroBoardCard;
