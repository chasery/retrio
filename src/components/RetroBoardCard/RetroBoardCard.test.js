import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RetroBoardCard from './RetroBoardCard';

describe('RetroBoardCard component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<RetroBoardCard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
