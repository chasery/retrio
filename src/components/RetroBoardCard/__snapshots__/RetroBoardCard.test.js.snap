// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetroBoardCard component renders without error 1`] = `
<li
  className="RetroBoardCard"
>
  <h4
    class="RetroBoardCard__headline"
  />
  <div
    class="RetroBoardCard__text"
  >
    <p />
  </div>
  <div
    class="RetroBoardCard__creator"
  >
    Submitted by 
    <strong />
  </div>
</li>
`;
